### PR TITLE
Vps 157/track-group-path

### PR DESCRIPTION
**Pull request by harbassan**
## Describe the issue

We need to track the path of a group within a scenario, so that when users open the scenario they are given the correct current scene that their group has reached.

## Describe the solution

Created a new API endpoint at `/group/path` with both a post and get request, the first of which adds a scene to the path and the second of which retrieves the most recent scene of that path. To make sure users don't overwrite each other, added simple checking and race condition handling using transactions.

## Risk

Could introduce conflicts with branch [VPS-154/csv-to-groups](https://github.com/UoaWDCC/VPS/tree/VPS-154/csv-to-groups) that will need to be resolved.

## Definition of Done

- [x] Code peer-reviewed
- [ ] Unit tests written and passing
- [ ] Integration tests written and passing
- [x] Continuous Integration build passing
- [x] Acceptance criteria met
- [x] Deployed to production environment

## Reviewed By

Jordan, Jin

### CODE REVIEW

**Review by JordanBlenn — APPROVED** [review]
Looks good to me - just wait for @wjin-lee to review before merging as he can critically assess the code. Good job guys!

File: backend/src/routes/api/group.js
Comment (wjin-lee) [inline backend/src/routes/api/group.js:24]: nice to see 409!

**Review by wjin-lee — APPROVED** [review]
lgtm! Nice job

**Pull request merged**
